packages/grafana-ui/src/components/Icon/iconMap.tsx

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+
+import { IconProps } from '@grafana/saga-icons';
+
+type IconImport = () => Promise<{ default: React.ComponentType<IconProps> }>;
+
+const dynamicIconImport = (iconName: string): IconImport => {
+  return () => import(/* webpackChunkName: "[request]" */ `@grafana/saga-icons/src/icons-gen/${iconName}`);
+};
+
+export const iconToComponentMap = {
```

Comment:
Will this make it more complicated for developers to add or remove Icons?

Reply:
This is mainly for backwards compatibility purposes. It's also needed for the icons that come from the server since they're sent as string names, but I think the number of those shouldn't be that large.